Spigot-Server-Patches/0212-Use-TerminalConsoleAppender-for-console-improvements.patch

```diff
@@ -0,0 +1,659 @@
++        minecraftserver.console = new com.destroystokyo.paper.console.TerminalConsoleCommandSender();
++        //minecraftserver.console = org.bukkit.craftbukkit.command.ColouredConsoleSender.getInstance();
++        //minecraftserver.reader.addCompleter(new org.bukkit.craftbukkit.command.ConsoleCommandCompleter(minecraftserver.server));
++        // Paper end
+         // CraftBukkit end
+ 
+         this.k = new GameProfileBanList(PlayerList.a);
+@@ -140,7 +143,7 @@ public abstract class PlayerList {
+ 
+         entityplayer.spawnIn(world);
+         entityplayer.setPosition(loc.getX(), loc.getY(), loc.getZ());
+-        entityplayer.setYawPitch(loc.getYaw(), loc.getPitch()); 
++        entityplayer.setYawPitch(loc.getYaw(), loc.getPitch());
+         // Spigot end
+ 
+         // CraftBukkit - Moved message to after join
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index d515ef0f..5a43f835 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -127,7 +127,7 @@ import io.netty.buffer.ByteBuf;
+ import io.netty.buffer.ByteBufOutputStream;
+ import io.netty.buffer.Unpooled;
+ import io.netty.handler.codec.base64.Base64;
+-import jline.console.ConsoleReader;
++// import jline.console.ConsoleReader; // Paper - JLine update
+ import org.apache.commons.lang.StringUtils;
+ import org.bukkit.NamespacedKey;
+ import org.bukkit.craftbukkit.util.CraftNamespacedKey;
+@@ -1088,9 +1088,11 @@ public final class CraftServer implements Server {
+         return logger;
+     }
+ 
+-    public ConsoleReader getReader() {
++    // Paper start - JLine update
++    /*public ConsoleReader getReader() {
+         return console.reader;
+-    }
++    }*/
++    // Paper end
+ 
+     @Override
+     public PluginCommand getPluginCommand(String name) {
+@@ -1390,7 +1392,7 @@ public final class CraftServer implements Server {
+ 
+         for (JsonListEntry entry : playerList.getProfileBans().getValues()) {
+             result.add(getOfflinePlayer((GameProfile) entry.getKey()));
+-        }        
++        }
+ 
+         return result;
+     }
+diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
+index 98e27fa2..68700f25 100644
+--- a/src/main/java/org/bukkit/craftbukkit/Main.java
++++ b/src/main/java/org/bukkit/craftbukkit/Main.java
+@@ -14,10 +14,11 @@ import java.util.logging.Logger;
+ import joptsimple.OptionParser;
+ import joptsimple.OptionSet;
+ import net.minecraft.server.MinecraftServer;
+-import org.fusesource.jansi.AnsiConsole;
++import net.minecrell.terminalconsole.TerminalConsoleAppender;
++//import org.fusesource.jansi.AnsiConsole; // Paper
+ 
+ public class Main {
+-    public static boolean useJline = true;
++    //public static boolean useJline = true; // Paper
```

Comment:
In effort to reduce diff, let's avoid commenting out stuff that won't impact behavior